File: LeamiServices/Database/Entities/Reservation.cs
using Leami.Model.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Leami.Services.Database.Entities
{
    public class Reservation
    {
        public int ReservationId { get; set; }
        public int UserId { get; set; }    
        public User? User { get; set; }  


        public DateOnly ReservationDate { get; set; }

        public TimeOnly ReservationTime { get; set; }

        public int NumberOfGuests { get; set; }

        public int ReservationStatus { get; set; }

        public string? ReservationReason { get; set; }

        public int? NumberOfMinors { get; set; } 

        public string ContactPhone { get; set; } = null!;

        public string? SpeciaLRequests { get; set; } 

    }
}

File: Leami.UI/leami_mobile/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:leami_mobile/models/user.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? reservationId;
  int? userId;
  String? reservationDate;
  String? reservationTime;

  int? numberOfGuests;
  int? reservationStatus;
  String? reservationReason;
  int? numberOfMinors;
  String? contactPhone;
  String? specialRequests;

  User? user;

  Reservation({
    this.reservationId,
    this.userId,
    this.reservationDate,
    this.reservationTime,

    this.numberOfGuests,
    this.reservationStatus,
    this.reservationReason,
    this.numberOfMinors,
    this.contactPhone,
    this.specialRequests,
    this.user,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);
}
